Sources/KituraContracts/CodableQuery/QueryDecoder.swift

```diff
@@ -56,7 +56,8 @@ public class QueryDecoder: Coder, Decoder {
      Initializer with a `[String : String]` dictionary.
      */
     public init(dictionary: [String : String]) {
-        self.dictionary = dictionary
+        let removedEmptydictionary = dictionary.filter { !$0.value.isEmpty }
+        self.dictionary = removedEmptydictionary
```

Comment:
This resolves the form decoding problem as referenced in the issue, but as you pointed out, it means that a struct encoded with QueryEncoder with a non-optional String value of `""` cannot be decoded.  This would be a regression and seems like a bad idea.

Really we want to only apply this interpretation to types for which `""` is not a valid representation.  Perhaps it could be along the lines of: try to decode, and if decoding fails, assign nil if it's an optional, otherwise raise a decoding error.